### PR TITLE
feat: add search.loading options for remote search

### PR DESCRIPTION
**Pull request by Timeless0911**
## Summary

- add search.loading options for remote search
- remove  1s of search index loading for local search

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by Timeless0911** [comment]
!canary

**Comment by Timeless0911** [comment]
!canary

**Comment by Timeless0911** [comment]
!canary

File: packages/shared/src/types/index.ts [hunk 1]
```diff
@@ -336,6 +336,7 @@ export type RemoteSearchOptions = SearchHooks & {
   domain?: string;
   indexName: string;
   searchIndexes?: RemoteSearchIndexInfo[];
+  searchLoading?: boolean;
```
Comment (chenjiahan) [inline packages/shared/src/types/index.ts:339]: should we document this new option?

Comment (Timeless0911) [reply packages/shared/src/types/index.ts:339]: remote search API is not stable and in long term will write a doc for it.

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**